### PR TITLE
Update README for development help

### PR DESCRIPTION
**Pull request by pepper-lees**
1. Included info on Docker being necessary
2. Updated instructions for modern devcontainer cli
3. Added general troubleshooting section (only one rn)

@ maintainers: Feel free to update the wording or placement of the additions/edits. I don't know how you'd like it to be organized.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3201,6 +3201,10 @@ For compatibility reasons, these NPM packages are embedded into bun’s binary a
 
 Estimated: 30-90 minutes :(
 
+### Prerequisites 
+
+- On all systems, [Docker](https://www.docker.com) is required. If using WSL on Windows, it is recommended to use [Docker Desktop](https://docs.microsoft.com/en-us/windows/wsl/tutorials/wsl-containers) for its WSL2 integration.
```
Comment (Jarred-Sumner) [inline README.md]: This isn't true. Docker is only required if using the devcontainer

Comment (pepper-lees) [reply README.md]: my bad, thanks for the correction! Fixing.

Comment (pepper-lees) [reply README.md]: Fixed, with some organization improvements too.

**Comment by Jarred-Sumner** [comment]
thank you

**Pull request merged**